### PR TITLE
Feature/header med søk

### PR DESCRIPTION
**Pull request by MengZhuNAV**
Storybook has not been configured because of a strange build error.
I will add it when I figured out a solution.

### CODE REVIEW

File: packages/familie-header/src/header/Header.tsx [hunk 1]
```diff
@@ -63,7 +63,7 @@ export const Bruker = ({ navn, enhet, popoverItems }: BrukerProps) => {
     );
 };
 
-const Header = ({ tittel, children, brukerinfo, tittelHref = '/', brukerPopoverItems }: HeaderProps) => {
+export const Header = ({ tittel, children, brukerinfo, tittelHref = '/', brukerPopoverItems }: HeaderProps) => {
```
Comment (henninghaakonsen) [inline packages/familie-header/src/header/Header.tsx]: I think we should move the search input component to its own package. This way we can easily add or remove the search from our clients.

File: packages/familie-header/README.md [hunk 2]
```diff
@@ -0,0 +1,11 @@
+# Header with Search Field
+
+A reusable page header component which includes a search field. It is developed for familie-ba-sak-frontend, but can be customized to fit other projects.
```
[the remark this answers is not in the document]
Comment (MengZhuNAV) [reply packages/familie-header/README.md]: Tried😅

File: packages/familie-header/src/header/Header.tsx [hunk 3]
```diff
@@ -33,24 +33,24 @@ interface BrukerProps {
 }
 
 export const Bruker = ({ navn, enhet, popoverItems }: BrukerProps) => {
-    const [anker, settAnker]= React.useState<HTMLElement|undefined>(undefined);
+    const [anker, settAnker] = React.useState<HTMLElement | undefined>(undefined);
```
[the remark this answers is not in the document]
Comment (MengZhuNAV) [reply packages/familie-header/src/header/Header.tsx]: Yes, it works as you wish😁

File: packages/familie-header/src/header/Header.tsx (hunk 1 above)
Comment (MengZhuNAV) [reply packages/familie-header/src/header/Header.tsx]: The search component is customizable, i.e. you can add or remove it as you wish:
<header>
  <søk/>
</header>

Comment (henninghaakonsen) [reply packages/familie-header/src/header/Header.tsx]: Awesome!

File: packages/familie-header/src/icons/IkonSpinner.tsx [hunk 4]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import {Ikon, IkonProps } from './Ikon';
+import './IkonSpinner.less';
+
+export const IkonSpinner = ({width = 12, height = 12 }: IkonProps) => {
```
[the remark this answers is not in the document]
Comment (MengZhuNAV) [reply packages/familie-header/src/icons/IkonSpinner.tsx]: That one is not very flexible. When I scale it to fit the header, it looks broken.

[the remark this answers is not in the document]
Comment (henninghaakonsen) [reply packages/familie-header/src/icons/IkonSpinner.tsx]: Ok, fair enough 👍

File: packages/familie-header/src/søk/Søk.tsx [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+    };
+
+    const prompt = !spinner && validator && gyldig && verdi.length > 0 && <IkonGyldig />
+        || !spinner && validator && !gyldig && verdi.length > 0 && <IkonFeil />
+
+    return (
+        <div className='søk_container'>
+            {prompt && <div className='søk_container__prompt'>{prompt}</div>}
+            {spinner && <div className='søk_container__spinner'><IkonSpinner /></div>}
+            <input className='søk_container__felt' onChange={innspillEndret} onKeyPress={tastenTrykkes} value={verdi} placeholder={plassholder} />
```
Comment (henninghaakonsen) [inline packages/familie-header/src/søk/Søk.tsx]: Sorry, I think we spoke past each other. In these cases I think we should keep onChange, onKeyPress and so forth :) But once there is something functional going on we can use norwegian (like søk)

Comment (MengZhuNAV) [reply packages/familie-header/src/søk/Søk.tsx]: I see. Actually there is a naming conflict here: the client send onChange as parameter and the innspillEndret will do something extra before triggering onChange.

**Pull request merged**